src/Http/Session.php

```diff
@@ -430,10 +430,11 @@ public function check(?string $name = null): bool
      * Returns given session variable, or all of them, if no parameters given.
      *
      * @param string|null $name The name of the session variable (or a path as sent to Hash.extract)
-     * @return string|array|null The value of the session variable, null if session not available,
+     * @param mixed $default The return value when the path does not exist
+     * @return mixed|null The value of the session variable, null if session not available,
```

Comment:
```suggestion
     * @return mixed The value of the session variable, null if session not available,
```

Reply:
No, that is on purpose. It must be |null - as you can see also on RFC passed for mixed in PHP8.
mixed does really not include null - as I have been telling you guys for years. The future PHP version will also enforce that.

Reply:
Oh darn, they merged the other one? https://wiki.php.net/rfc/mixed_type_v2
Where mixed is basically `?mixed`? That's sad...

Reply:
They really screwed this up :(
https://wiki.php.net/rfc/union_types_v2 also is in line what should have been `?mixed`...

Reply:
As long as there is still a chance for them to fix this, I would suggest we keep the proper nullable declarations in place - as this can easily be cleaned up by tooling then either way as long as we have the additional info in place. Losing it now makes it harder to revive.
Also, this is still way clearer in terms of nullable communication if we adhere to this verbosity.

Reply:
I don't know about php union types, but I do prefer to have some indication of nullable in the docblock types.

Reply:
Okay, you guys can have it your way until static analyzers or PHP itself starts complaining about it :slightly_smiling_face:.

Reply:
I agree mixed a difficult notation ... but just for consistency until we find out what blows up in the php 8 analyzers.